backend/fms_core/services/samplesheet.py

```diff
@@ -60,22 +56,224 @@ def get_samplesheet(container_kind, placement):
                         index = Index.objects.get(id=index_id)
                         sequences_3prime = ", ".join(index.list_3prime_sequences)
                         sequences_5prime = ", ".join(index.list_5prime_sequences)
-                        row_data_by_lane[lane].append({COLUMN_LANE: lane, COLUMN_SAMPLE_ALIAS: sample_alias, COLUMN_INDEX_3PRIME: sequences_3prime, COLUMN_INDEX_5PRIME: sequences_5prime})
+                        row_data_by_lane.append((str(lane), sample_alias, sequences_3prime, sequences_5prime))
                     else:
                         errors.append(f'Cannot find index associated to sample {sample_alias}.')
-            ordered_lanes = sorted(row_data_by_lane.keys())
-            i = 1
-            for lane in ordered_lanes:
-                for row_data in row_data_by_lane[lane]:
-                    sheet.cell(row=SAMPLESHEET_FIRST_INPUT_ROW + i, column=COLUMN_LANE).value = str(row_data[COLUMN_LANE])
-                    sheet.cell(row=SAMPLESHEET_FIRST_INPUT_ROW + i, column=COLUMN_SAMPLE_ALIAS).value = row_data[COLUMN_SAMPLE_ALIAS]
-                    sheet.cell(row=SAMPLESHEET_FIRST_INPUT_ROW + i, column=COLUMN_INDEX_3PRIME).value = row_data[COLUMN_INDEX_3PRIME]
-                    sheet.cell(row=SAMPLESHEET_FIRST_INPUT_ROW + i, column=COLUMN_INDEX_5PRIME).value = row_data[COLUMN_INDEX_5PRIME]
-                    i = i + 1
         else:
             errors.append(f"Container of kind {container_kind} cannot be used for an experiment run.")
     except Exception as err:
         errors.append(err)
 
+    row_data_by_lane.sort(key=lambda x: x[0])
+    bclconvert_data = [
+        BCLConvert_Datum(Lane=lane, Sample_ID=sample_alias, Index=sequences_3prime, Index2=sequences_5prime)
```

Comment:
how about using "*" as a separator and have Sample_ID be sample_alias + "*" + derived_sample_id ? Both here and for germline. The idea is to make the sample_ID unique while providing an easy separator for downstream processing to remove the part they don't want ...